subsys/bluetooth/host/hci_core.c

```diff
@@ -3779,10 +3790,30 @@ int bt_disable(void)
 
 	bt_monitor_send(BT_MONITOR_CLOSE_INDEX, NULL, 0);
 
+#if defined(CONFIG_BT_PRIVACY)
+	k_work_cancel_delayable(&bt_dev.rpa_update);
+#endif
+
 #if defined(CONFIG_BT_PER_ADV_SYNC)
 	bt_periodic_sync_disable();
 #endif /* CONFIG_BT_PER_ADV_SYNC */
 
+#if defined(CONFIG_BT_CONN)
+	bt_pub_key_hci_disrupted();
+	bt_conn_cleanup_all_conns();
+	disconnected_handles_reset();
+
+#if defined(CONFIG_BT_L2CAP_DYNAMIC_CHANNEL)
+	bt_l2cap_server_unregister_all();
+	bt_l2cap_free_tx_context();
+#endif /* CONFIG_BT_L2CAP_DYNAMIC_CHANNEL */
+#endif /* CONFIG_BT_CONN */
+
+	err = bt_id_delete_all();
+	if (err) {
+		return err;
+	}
```

Comment:
I understand why it's done like this, but basically we here return an err on `bt_disable`, indicating that it failed, but we have actually disable almost everything at this point. I wonder if this should be replaced with an assert instead? It can't be recovered in any case.